storm/src/main/storm/mesos/resources/ResourceEntries.java

```diff
@@ -93,15 +93,15 @@ public ReservationType getReservationType() {
     public RangeResourceEntry add(ResourceEntry<Long> resourceEntry) {
```

Comment:
> please try to answer these "asks" from @JessicaLHartog :
Tried to address them via unit tests. Them being the following 
* Before we continue this discussion, can you please provide some comments and/or an example for how this code is supposed to behave?
* It would probably be most beneficial for my understanding if you provide this explanation without looking at the code aside from the signature provided here

Initially the code looked a bit confusing. The latest code code looks pretty straightforward and tests demonstrate the usage. Looks like you guys are in favor of comments. So let me go ahead and add them anyway.